lib/Ast_to_Mir.ml

```diff
@@ -64,11 +59,21 @@ let truncate_dist ast_obs t =
          (BinOp (obs, cond, trans_expr x), bind_loc x.expr_typed_loc add_inf, y))
   in
   match t with
-  | Ast.NoTruncate -> None
-  | Ast.TruncateUpFrom lb -> Some (trunc Less lb None)
-  | Ast.TruncateDownFrom ub -> Some (trunc Greater ub None)
+  | Ast.NoTruncate -> []
+  | Ast.TruncateUpFrom lb -> [trunc Less lb None]
+  | Ast.TruncateDownFrom ub -> [trunc Greater ub None]
   | Ast.TruncateBetween (lb, ub) ->
-      Some (trunc Less lb (Some (trunc Greater ub None)))
+      [trunc Less lb (Some (trunc Greater ub None))]
+
+let unquote s =
+  if s.[0] = '"' && s.[String.length s - 1] = '"' then
+    String.drop_suffix (String.drop_prefix s 1) 1
+  else s
+
+let trans_printable (p : Ast.typed_expression Ast.printable) =
+  match p with
+  | Ast.PString s -> Lit (Str, unquote s)
+  | Ast.PExpr e -> trans_expr e
 
 let rec trans_stmt {Ast.stmt_typed; stmt_typed_loc; _} =
   let or_skip = Option.value ~default:Skip in
```

Comment:
In the operator assignment below, you are treating a binary operator as a function application. Before, it seemed like binary operators were not treated as functions. What is the reason for that inconsistency?

Also, why is the lhs of an assignment treated as a plain expression?

Reply:
So obviously a var or indexed thingy are expressions, so that works out, but of course you can put nonsense stuff in the MIR as the lhs of assignment if it's just "expressions." I think @rybern  suggested making an LHS type in the MIR somewhere that would fix this. I think I would also like to just remove the concept of Indexed as an Expr from the MIR completely, to be replaced with an explicit For loop. Then we'd just have to use string identifiers everywhere we wanted to just refer to a var, like here on the lhs of assignment.

> In the operator assignment below, you are treating a binary operator as a function application. Before, it seemed like binary operators were not treated as functions. What is the reason for that inconsistency?

It would help me if you put the PR comments below the code in question as github shows only the code directly above the comment. And yeah, this is left over from when I was just treating binops as fnapps. We should probably have a larger discussion about that topic - I'm kind of ambivalent either way and would sort of rather it just be simpler if we can get away with it.

Reply:
> It would help me if you put the PR comments below the code in question as github shows only the code directly above the comment.
I know, sorry. I wanted to comment on something that wasn't new in this PR and github wouldn't let me comment on the actual line that I wanted to comment on. It will only let me comment on lines that were changed in this PR.

Reply:
> I think I would also like to just remove the concept of Indexed as an Expr from the MIR completely, to be replaced with an explicit For loop.

Not sure I understand how that would work. Can you please explain? We are in a language with arrays, right? Don't you need indexing? Or do you have a way in mind to treat arrays as syntactic sugar?

Reply:
We talked about this in person, but just to record - the Single index we still need something like, but we could represent all of the multi indexing as For loops. But for now, since the target languages have more efficient ways of dealing with multi indexing than just for loops and we don't want to take the chance that it's not as performant without an optimization pass, we'll leave them in.